homeassistant/components/devolo_home_control/binary_sensor.py

```diff
@@ -50,10 +50,21 @@ class DevoloBinaryDeviceEntity(DevoloDeviceEntity, BinarySensorEntity):
 
     def __init__(self, homecontrol, device_instance, element_uid):
         """Initialize a devolo binary sensor."""
-        if device_instance.binary_sensor_property.get(element_uid).sub_type != "":
-            name = f"{device_instance.itemName} {device_instance.binary_sensor_property.get(element_uid).sub_type}"
+        self._binary_sensor_property = device_instance.binary_sensor_property.get(
+            element_uid
+        )
+
+        self._device_class = DEVICE_CLASS_MAPPING.get(
+            self._binary_sensor_property.sub_type
+            or self._binary_sensor_property.sensor_type
+        )
+        if self._device_class is None:
+            if device_instance.binary_sensor_property.get(element_uid).sub_type != "":
+                name = f"{device_instance.itemName} {device_instance.binary_sensor_property.get(element_uid).sub_type}"
+            else:
+                name = f"{device_instance.itemName} {device_instance.binary_sensor_property.get(element_uid).sensor_type}"
         else:
-            name = f"{device_instance.itemName} {device_instance.binary_sensor_property.get(element_uid).sensor_type}"
+            name = device_instance.itemName
```

Comment:
same

Reply:
Thanks for the hint. Changed in latest commit.